xds/src/main/java/io/grpc/xds/EdsLoadBalancer.java

```diff
@@ -0,0 +1,737 @@
+    return true;
+  }
+
+  @Override
+  public void shutdown() {
+    channelLogger.log(ChannelLogLevel.DEBUG, "EDS load balancer is shutting down");
+
+    if (endpointWatcher != null) {
+      xdsClient.cancelEndpointDataWatch(edsServiceName, endpointWatcher);
+      endpointWatcher.reset();
+    }
+  }
+
+  static final class EdsConfig {
+
+    /**
+     * Name of cluster to query EDS for.
+     */
+    final String name;
+
+    final Object localityPickingPolicy;
+
+    EdsConfig(String name, Object localityPickingPolicy) {
+      this.name = checkNotNull(name, "name");
+      this.localityPickingPolicy = checkNotNull(localityPickingPolicy);
+    }
+  }
+
+  static final class EndpointWatcherImpl implements EndpointWatcher {
+
+    private static final String ROUND_ROBIN = "round_robin";
+    private static final long DELAYED_DELETION_TIMEOUT_MINUTES = 15L;
+    private static final EndpointWatcherImpl.PickerFactory pickerFactoryImpl =
+        new EndpointWatcherImpl.PickerFactory() {
+          @Override
+          public SubchannelPicker picker(List<WeightedChildPicker> childPickers) {
+            return new InterLocalityPicker(childPickers);
+          }
+        };
+
+    private final Helper helper;
+    private final PickerFactory pickerFactory;
+    private final LoadBalancerProvider loadBalancerProvider;
+    private final ThreadSafeRandom random;
+    private final LoadStatsStore loadStatsStore;
+    private final OrcaPerRequestUtil orcaPerRequestUtil;
+    private final OrcaOobUtil orcaOobUtil;
+    private final PriorityManager priorityManager = new PriorityManager();
+    private final Map<Locality, LocalityLbInfo> localityMap = new HashMap<>();
+    // Most current set of localities instructed by traffic director
+    private Set<Locality> localities = ImmutableSet.of();
+    private List<DropOverload> dropOverloads = ImmutableList.of();
+    private long metricsReportIntervalNano = -1;
+
+    EndpointWatcherImpl(
+        Helper helper,
+        PickerFactory pickerFactory,
+        LoadBalancerRegistry lbRegistry,
+        ThreadSafeRandom random,
+        LoadStatsStore loadStatsStore,
+        OrcaPerRequestUtil orcaPerRequestUtil,
+        OrcaOobUtil orcaOobUtil) {
+      this.helper = checkNotNull(helper, "helper");
+      this.pickerFactory = checkNotNull(pickerFactory, "pickerFactory");
+      loadBalancerProvider = checkNotNull(
+          lbRegistry.getProvider(ROUND_ROBIN),
+          "Unable to find '%s' LoadBalancer", ROUND_ROBIN);
+      this.random = checkNotNull(random, "random");
+      this.loadStatsStore = checkNotNull(loadStatsStore, "loadStatsStore");
+      this.orcaPerRequestUtil = checkNotNull(orcaPerRequestUtil, "orcaPerRequestUtil");
+      this.orcaOobUtil = checkNotNull(orcaOobUtil, "orcaOobUtil");
+    }
+
+    @Override
+    public void onEndpointChanged(EndpointUpdate update) {
+      checkNotNull(update, "update");
+
+      updateDropPercentage(ImmutableList.copyOf(update.getDropPolicies()));
```

Comment:
Inline your implementation here. You have too many simplex methods one calling another. When you migrate, you are only reusing implementation logic, not necessarily class structure. You should restructure methods if they are inappropriate.